protocols/gossipsub/src/handler.rs

```diff
@@ -571,10 +529,37 @@ impl ConnectionHandler for Handler {
             ConnectionEvent::FullyNegotiatedOutbound(fully_negotiated_outbound) => {
                 self.on_fully_negotiated_outbound(fully_negotiated_outbound)
             }
-            ConnectionEvent::DialUpgradeError(DialUpgradeError { error: e, .. }) => {
+            ConnectionEvent::DialUpgradeError(DialUpgradeError { error, .. }) => {
                 self.outbound_substream_establishing = false;
-                warn!("Dial upgrade error {:?}", e);
-                self.upgrade_errors.push_back(e);
+
+                match error {
+                    // Timeout errors get mapped to NegotiationTimeout and we close the connection.
+                    ConnectionHandlerUpgrErr::Timeout | ConnectionHandlerUpgrErr::Timer => {
+                        self.keep_alive = KeepAlive::No;
+                        log::info!("Dial upgrade error: {}", HandlerError::NegotiationTimeout);
+                    }
+                    // There was an error post negotiation, close the connection.
+                    ConnectionHandlerUpgrErr::Upgrade(UpgradeError::Apply(e)) => {
+                        self.keep_alive = KeepAlive::No;
```

Comment:
This isn't fatal, lets not set `keep_alive` here but just log the error.

Reply:
addressed